Add mock for core module (#381)
DataDog/dd-sdk-reactnative: docs/advanced_configuration.md

If you have not set up the SDK yet, follow the [in-app setup instructions][1] or refer to the [React Native RUM setup documentation][2].

> ## Testing with Jest

Author: This part is needed as soon as you've done the basic set-up, before going further in implementation.
Customers will want to make sure their tests are passing before going implementing more features, so they can test their implementation along the way. 
I think it will also be more useful than each individual guides for custom instrumentations and initializations.

> Testing apps using `'@datadog/mobile-react-native'` may require some extra steps since Native Modules are non existent in testing environments.

Reviewer: ```suggestion
Testing apps using `'@datadog/mobile-react-native'` might require completing extra steps, since Native Modules do not exist in testing environments.
```

> We provide mocks for the `'@datadog/mobile-react-native'` package. To use them with [Jest][jest], add the following in your Jest setup file:

Reviewer: ```suggestion
Datadog provides mocks for the `'@datadog/mobile-react-native'` package. To use them with [Jest][4], add the following in your Jest setup file:
```

```javascript
jest.mock('@datadog/mobile-react-native', () => {
    return require('@datadog/mobile-react-native/jest/mock');
});
```

> This disables interaction, error and resource automated tracking if you use the `DatadogProvider`.

Reviewer: ```suggestion
Use `DatadogProvider` to disable automated tracking for interactions, errors, and resources.
```

Reviewer: Let me know if I changed the meaning of the sentence.

Reviewer: The change looks good. Just a small nit, can we change "will be" to "is"?
So:
> Interaction, error and resource automated tracking is disabled in your tests if you initialize the SDK with the `DatadogProvider` component.

All SDK methods are mocked by `jest.fn()`, so you can assert that a Datadog SDK method was called:

```javascript
import { DdLogs } from '@datadog/mobile-react-native';

describe('App', () => {
    it('calls DdLogs.debug on mount', () => {
        renderer.create(<App />);
        expect(DdLogs.debug).toHaveBeenCalledWith('app started');
    });
});
```

> If you use an other test runner than Jest, you will need to create the mocks for your test runner.

Reviewer: ```suggestion
If you use a test runner other than Jest, you needs to create the mocks for your test runner.
```

Author: ```suggestion
If you use a test runner other than Jest, you need to create the mocks for your test runner.
```

[1]: https://app.datadoghq.com/rum/application/create
[2]: https://docs.datadoghq.com/real_user_monitoring/reactnative
[3]: https://reactnative.dev/docs/interactionmanager#runafterinteractions

> [jest]: https://jestjs.io/

Reviewer: ```suggestion
[4]: https://jestjs.io/
```
